libc-top-half/musl/arch/wasm32/atomic_arch.h

```diff
@@ -4,3 +4,12 @@
 #define a_clz_64 __builtin_clzll
 #define a_ctz_32 __builtin_ctz
 #define a_ctz_64 __builtin_ctzll
+
+#ifdef __wasm64__
+#define a_cas_p a_cas_p
+static inline void *a_cas_p(volatile void *p, void *t, void *s)
```

Comment:
Why is this needed for __wasm64__ but not for __wasm32__?

Reply:
because a_cas_p is defined by the clang front end for wasm32 but wasm64

Reply:
`c_cas_p` doesn't sounds like something that clang itself would ever define...

Reply:
well, the problem is that musl does not need c_cas_p on 32 bit platform, but only on 64 bit platform and i am not 100% confident of the correctness here.

Reply:
fixed